src/libopensc/pkcs15-cert.c

```diff
@@ -141,6 +133,195 @@ parse_x509_cert(sc_context_t *ctx, struct sc_pkcs15_der *der, struct sc_pkcs15_c
 }
 
 
+/* get a component of a Distinguished Name (e.i. subject or issuer) from an oid tag. dn can be either
+ * cert->subject or cert->issuer. dn_len would be cert->subject_len or cert->issuer_len.
+ * common types:
+ *   CN: type[] = {0x55, 0x04, 0x03} type_len = 3;
+ *   Country: type[] = {0x55, 0x04, 0x06} type_len = 3;
+ *   L: type[] = {0x55, 0x04, 0x07} type_len = 3;
+ *   S: type[] = {0x55, 0x04, 0x08} type_len = 3;
+ *   O: type[] = {0x55, 0x04, 0x0a} type_len = 3;
+ *   OU: type[] = {0x55, 0x04, 0x0b} type_len = 3;
```

Comment:
The comments should be changed to use sc_object_id way to list a OID.  
For example:
  struct sc_object_id keyUsage_oid={{2,5,29,15,-1}};